src/XmlSerializationVisitor.php

```diff
@@ -491,4 +500,16 @@ private function getClassDefaultNamespace(ClassMetadata $metadata): ?string
     {
         return $metadata->xmlNamespaces[''] ?? null;
     }
+
+    private function parentHasNonEmptyDefaultNs(): bool
+    {
+        $element = $this->currentNode;
+        while ($element) {
```

Comment:
Looks good so far. I'm just worried about the performance impact of this loop...
It should be tested carefully

Reply:
The impact should not be too severe, as it will only be triggered if the element to be serialized has the explicit `''` (empty string) namespace and it will only loop up until it hits any `xmlns="..."` (non empty default namespace).

I am open to suggestions of how to test this performance wise.

Reply:
I have tried to debug this and IMO should be enough.

```php
return (null !== ($uri = $this->currentNode->lookupNamespaceUri(null)) && ('' !== $uri));
```

Consider that `lookupNamespaceUri` is aware of the position of the dom element and will give you the default namespace for a node considering its parents.

See https://3v4l.org/ONH44

Reply:
I think you are right. I was thinking that I have to traverse the tree up but we can scrap the loop.

So I will drop the loop but keep the method?

Reply:
Sounds good to me. We can keep the method for readability